src/vs/base/browser/ui/toggle/toggle.ts

```diff
@@ -246,9 +248,19 @@ export class Toggle extends Widget {
 		this.domNode.classList.add('disabled');
 	}
 
-	setTitle(newTitle: string): void {
+	setTitle(newTitle: string | IMarkdownString | HTMLElement): void {
 		this._title = newTitle;
-		this.domNode.setAttribute('aria-label', newTitle);
+
+		let ariaLabel = '';
+		if (typeof newTitle === 'string') {
+			ariaLabel = newTitle;
+		} else if (hasKey(newTitle, { value: true })) {
+			ariaLabel = newTitle.value;
+		} else if (hasKey(newTitle, { textContent: true })) {
```

Comment:
at this point we can then assume HTMLElement... or if you really wanna be clear, there's

https://github.com/microsoft/vscode/blob/0551264cd7c263b02c18940126555b0acab6cfab/src/vs/base/browser/dom.ts#L1012-L1015